sdks/python/apache_beam/io/gcp/bigquery_tools.py

```diff
@@ -182,7 +182,10 @@ def parse_table_reference(table, dataset=None, project=None):
   """
 
   if isinstance(table, bigquery.TableReference):
-    return table
+    return bigquery.TableReference(
```

Comment:
Probably update the spec above to mention that a new object will be returned ?

Reply:
Another option might be to create a new object at line 343 which might be cleaner

Reply:
Done.

Reply:
Ah... ok.. I can do that second option

Reply:
on second thought, I think it's a better practice to avoid reusing objects, and create copies for clarity...